Assets/Scripts/UI/Components/UIButtonLogic.cs

```diff
@@ -10,27 +11,48 @@ namespace Netherlands3D.Twin
 {
     public class UIButtonLogic : MonoBehaviour
     {
-
+        private Button button;
+        
         public Color BaseTextColor = Color.blue;
         public Color HighlightedTextColor = Color.white;
         public Color DisabledTextColor = Color.grey;
         public TextMeshProUGUI ButtonText;
-        public GameObject Icon;
+        public Image Icon;
 
-        // Start is called before the first frame update
-        void Start()
+        private void Awake()
         {
-            if (ButtonText != null)
+            button = GetComponent<Button>();
+        }
+
+        private void Start()
+        {
+            if (button.interactable)
             {
-                ButtonText.overrideColorTags = true;
-                ButtonText.color = BaseTextColor;
+                if (ButtonText != null)
+                {
+                    ButtonText.overrideColorTags = true;
+                    ButtonText.color = BaseTextColor;
+                }
+                if (Icon != null)
+                {
+                    Icon.color = BaseTextColor;
+                }
             }
-            if(Icon != null)
+            else
             {
-                Icon.GetComponent<Image>().color = BaseTextColor;
+                if (ButtonText != null)
+                {
+                    ButtonText.overrideColorTags = true;
+                    ButtonText.color = DisabledTextColor;
+                }
+                if (Icon != null)
+                {
+                    Icon.color = DisabledTextColor;
+                }
             }
```

Comment:
dit stuk lijk je 2 keer te gebruiken ook in pointerup, kan je misschien gewoon een method maken zoals,

private void EnableButton()
{
  bool interactable = button.interactable;
  Color color = interactable ? BaseTextColor : DisabledTextColor;
  if (ButtonText != null)
  {
      ButtonText.overrideColorTags = true;
      ButtonText.color = color;
  }
  if (Icon != null)
  {
      Icon.color = color;
  }
}